src/ui/app/components/transactionBuilder.jsx

```diff
@@ -375,6 +404,109 @@ const TransactionBuilder = React.forwardRef(({ onConfirm }, ref) => {
         }
         ref={withdrawRef}
       />
+      <ConfirmModal
+        ready={data.ready}
+        title="Stake deregistration"
+        sign={async (password, hw) => {
+          if (hw)
+            return await signAndSubmitHW(data.tx, {
+              keyHashes: [
+                data.account.paymentKeyHash,
+                data.account.stakeKeyHash,
+              ],
+              account: data.account,
+              hw,
+            });
+          return await signAndSubmit(
+            data.tx,
+            {
+              keyHashes: [
+                data.account.paymentKeyHash,
+                data.account.stakeKeyHash,
+              ],
+              accountIndex: data.account.index,
+            },
+            password
+          );
+        }}
+        onConfirm={(status, signedTx) => {
+          if (status === true)
+            toast({
+              title: 'Deregistration submitted',
+              status: 'success',
+              duration: 4000,
+            });
+          else if (signedTx === ERROR.fullMempool) {
+            toast({
+              title: 'Transaction failed',
+              description: 'Mempool full. Try again.',
+              status: 'error',
+              duration: 3000,
+            });
+          } else
+            toast({
+              title: 'Transaction failed',
+              status: 'error',
+              duration: 3000,
+            });
+          undelegateRef.current.closeModal();
+        }}
+        info={
+          <Box
+            width="100%"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            flexDirection="column"
+          >
+            <Icon as={GoStop} w={50} h={50} color="red.500" />
+            <Box h="4" />
+            <Text fontSize="sm">
+              Going forward with deregistration will have the following effects:
+            </Text>
+            <UnorderedList mt="10px">
+              <ListItem>You will no longer receive rewards</ListItem>
```

Comment:
```suggestion
              <ListItem>You will no longer receive rewards.</ListItem>
```

I’m not sure if punctuation helps here. I would assume it does, but fee free to ignore this suggestion!

Reply:
(same with the other list items)